calc/index.php

```diff
@@ -14,14 +14,14 @@
 <body>
 
 <form action="/calc/index.php" method="post">
-    <input type="number" name="x" value="<?php echo $x; ?>">
+    <input type="number" name="x" value="<?php echo $x; ?>" step="any">
     <select name="op">
         <option value="+"<?php if ('+'==$op) { ?> selected <?php } ?>>+</option>
         <option value="-"<?php if ('-'==$op) { ?> selected <?php } ?>>-</option>
         <option value="*"<?php if ('*'==$op) { ?> selected <?php } ?>>*</option>
         <option value="/"<?php if ('/'==$op) { ?> selected <?php } ?>>/</option>
     </select>
```

Comment:
Try to optimize building of the options and make it more dynamicaly.
Tip (foreach and arrays needs).